src/main/kotlin/cacophony/grammars/ParseTree.kt

```diff
@@ -8,11 +8,24 @@ sealed class ParseTree<SymbolType : Enum<SymbolType>>(
 ) {
     class Leaf<SymbolType : Enum<SymbolType>>(
```

Comment:
intellij tells me that if it has custom `equals`, it should also have `hashCode`

Reply:
Why?